### DISCUSSION TITLE
About openxlsx2 dims

### DISCUSSION THREAD

**Discussion by olivroy** [emitanaka/edibble#65] [General]
I saw that you are now using openxlsx2 in your pkg.

https://github.com/emitanaka/edibble/blob/4ab9381309bfd993df29c913583499c61f0c35be/R/export.R#L291

`wb_dims(1:nrow(data) + 1L, jdata)`

You want to select a column in the data right?

this can be replaced by the handy syntax
```r
wb_dims(x = dat, cols = jdata, select = "x")

# select = "x" default when specifying the full data
# select = "data" to select only the data grid
# select = "col_names" to style column names
# select = "row_names" (when they exist) to style row names
```

If there is something in the docs we can do to make this more obvious!

Feel free to disregard if you don't think it is relevant for you! Just wanted to give a heads-up in case this special syntax is useful for you.

Basically using `wb_dims()` with `x` makes sure you keep track of the importance of the data you added when you put style afterwards.

Ref: https://janmarvin.github.io/openxlsx2/reference/wb_dims.html

I also made a little demo in a blog
https://[olivroy.quarto.pub/creating-cool-xlsx-files-with-openxlsx2-10/](https://olivroy.quarto.pub/creating-cool-xlsx-files-with-openxlsx2-10/)

**Comment by emitanaka** [comment]
Thanks @olivroy , I think it doesn't apply in my case because I am not using a data frame. Ta!
